tests/models/test_baseoperator.py

```diff
@@ -563,3 +564,11 @@ def test_set_xcomargs_dependencies_error_when_outside_dag(self):
         with pytest.raises(AirflowException):
             op1 = DummyOperator(task_id="op1")
             CustomOp(task_id="op2", field=op1.output)
+
+    def test_replace_dummy_trigger_rule(self):
+        with DAG(dag_id='dummy_replace', default_args={"start_date": datetime.today()}):
+            op1 = DummyOperator(task_id="op1", trigger_rule="dummy")
+            op2 = DummyOperator(task_id="op2", trigger_rule=TriggerRule.DUMMY)
```

Comment:
```suggestion
            with pytest.warns(DeprecationWarning):
                op1 = DummyOperator(task_id="op1", trigger_rule="dummy")
            with pytest.warns(DeprecationWarning):
                op2 = DummyOperator(task_id="op2", trigger_rule=TriggerRule.DUMMY)
```

Should we make sure the `DeprecationWarning` happens (maybe even with a `match` to check the message too)?

Reply:
thanks! will add